fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/patient/ChunkReader.java

```diff
@@ -108,73 +126,88 @@ public ChunkReader() {
+                    queryParameters.put(Constants.FHIR_SEARCH_LASTUPDATED, searchCriteria);
                 }
 
                 queryParameters.put("_sort", Arrays.asList(new String[] { Constants.FHIR_SEARCH_LASTUPDATED }));
-                searchContext = SearchUtil.parseQueryParameters("Patient", patient.getId(),
-                        ModelSupport.getResourceType(resourceTypes.get(indexOfCurrentResourceType)), queryParameters, true);
-                do {
-                    searchContext.setPageSize(pageSize);
-                    searchContext.setPageNumber(compartmentPageNum);
-                    FHIRTransactionHelper txn = new FHIRTransactionHelper(fhirPersistence.getTransaction());
-                    txn.begin();
-                    FHIRPersistenceContext persistenceContext = FHIRPersistenceContextFactory.createPersistenceContext(null, searchContext);
-                    List<Resource> resources2 = fhirPersistence.search(persistenceContext, resourceType).getResource();
-                    txn.commit();
-                    compartmentPageNum++;
-
-                    for (Resource res2 : resources2) {
-                        if (res2 == null) {
-                            continue;
-                        }
-                        try {
-                            FHIRGenerator.generator(Format.JSON).generate(res2, chunkData.getBufferStream());
-                            chunkData.getBufferStream().write(Constants.NDJSON_LINESEPERATOR);
-                            resSubTotal++;
-                        } catch (FHIRGeneratorException e) {
-                            if (res.getId() != null) {
-                                logger.log(Level.WARNING, "fillChunkDataBuffer: Error while writing resources with id '"
-                                        + res.getId() + "'", e);
-                            } else {
-                                logger.log(Level.WARNING,
-                                        "fillChunkDataBuffer: Error while writing resources with unknown id", e);
+
+                for (String patientId : patientIds) {
+
+                    searchContext = SearchUtil.parseQueryParameters("Patient", patientId,
+                            ModelSupport.getResourceType(resourceTypes.get(indexOfCurrentResourceType)), queryParameters, true);
+                    do {
+                        searchContext.setPageSize(pageSize);
+                        searchContext.setPageNumber(compartmentPageNum);
+                        FHIRTransactionHelper txn = new FHIRTransactionHelper(fhirPersistence.getTransaction());
+                        txn.enroll();;
+                        FHIRPersistenceContext persistenceContext = FHIRPersistenceContextFactory.createPersistenceContext(null, searchContext);
+                        List<Resource> resources = fhirPersistence.search(persistenceContext, resourceType).getResource();
+                        txn.unenroll();;
```

Comment:
```suggestion
                        txn.unenroll();
```

Reply:
good catch, my eclipse has rendering issue, I have to resize the editor window the show the correct contents sometime, seems I missed these, thanks! considering to upgrade my eclipse now, I'm still using 2019 June version ...